### PR TITLE
feat: prebuilt

### LINKED ISSUE #25: Prebuilt and static link
Hi, thank you for creating this library.
I would like to ask if there are plans to provide a static link version and a pre-built download of the library. So that users can use it without installing vspkg. Just like [rust-skia](https://github.com/rust-skia/rust-skia).
I can contribute this part of the code if you want.

### PR DESCRIPTION
**Pull request by EYHN**
close #25 

This pr adds prebuilt feature to `russimp-sys`, When the prebuilt feature is enabled `russimp-sys` will not depend on the system wide `assimp` installation, at the cost of a larger compiled executable. Please see the changed [readme file](https://github.com/EYHN/russimp-sys#prebuilt-1) for details. 

Prebuilt results can be viewed here
https://github.com/EYHN/russimp-sys/actions/runs/1908424353
https://github.com/EYHN/russimp-sys/releases

# TODO
* [ ]  I added `assimp` as a submodule and checkedout to `v5.2.2`. Does `russimp` need any changes for this? Although I tested all of them and all passed.
* [x] documents
* [ ] Change the download url in build.rs when everything done

test:
* [x] macos
* [x] linux
* [x] windows

# Maintenance Guide

!!! The new action requires a secret named `ACCESS_TOKEN` with a personal access token with `repo` scopes access, created from here https://github.com/settings/tokens/new

When releasing a version, first change the version in Cargo.toml.

Then use git to create a tag with the version number, action will run to compile pre-built packages for each platform and upload them to github releases.

``` bash
git tag v1.0.1
git push origin v1.0.1
```

Releases are saved as drafts so no one else can see them, so don't forget to click publish release when you're done.

Then manually publish to crates.io. Publishing is irreversible, so I think it's better to do it manually.

### CODE REVIEW

**Comment by EYHN** [comment]
image

I have tested it works fine on windows

**Comment by jkvargas** [comment]
hey man,
Is it possible to have static link as an option?
I guess would be nice to have the user to decide if he wants to use the library statically linked with prebuilt assimp or use the one available on his own system.
Regarding 5.2.2, I am not sure if it breaks user space. Anyways, on russimp side I increase its version everytime russimp-sys is modified.

**Comment by jkvargas** [comment]
hey man, sorry for the delay answering you.
Lets make system libraries by default. Since this is the current default.

**Comment by EYHN** [comment]
https://github.com/jkvargas/russimp-sys/pull/26/files#diff-d0d98998092552a1d3259338c2c71e118a5b8343dd4703c0c7f552ada7f9cb42R348-R352

Please take a look at the latest changes, uses the system assimp library by default

**Comment by jkvargas** [comment]
thanks for the contribution bud.
Soon as I leave the office, Today, I will review it :)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+name: Build Release
```
Comment (jkvargas) [inline .github/workflows/release.yml:1]: hey man,
I've seen that you removed the former github action but I am not seeing the cargo publish step.

File: Cargo.toml [hunk 2]
```diff
@@ -10,21 +10,24 @@ categories = ["rendering", "external-ffi-bindings", "game-engines", "multimedia"
...
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+exclude = ["/assimp"]
 
 [lib]
 name = "russimp_sys"
 path = "src/lib.rs"
 
-[build-dependencies]
-bindgen = "0.59.1"
-vcpkg = "0.2.15"
```
Comment (jkvargas) [inline Cargo.toml:22]: I'd rather keep vcpkg as an option as well.
vcpkg has a pull request to update assimp to the latest version

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        target:
+          - os: ubuntu-20.04
+            triple: x86_64-unknown-linux-gnu
+          - os: macos-11
+            triple: x86_64-apple-darwin
+          - os: macos-11
+            triple: aarch64-apple-darwin
+          - os: windows-2019
+            triple: x86_64-pc-windows-msvc
+        features: ['static-link', 'static-link,nozlib']
```
Comment (jkvargas) [inline .github/workflows/release.yml:36]: I believe dynamic linking with the system libraries should be the default option, the one that is going to be published to crates.io
additionally we can have a release build with assimp compiled and statically linked :)

File: src/lib.rs [hunk 4]
```diff
@@ -2,6 +2,7 @@
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
 #![allow(improper_ctypes)]
+#![allow(deref_nullptr)]
```
Comment (jkvargas) [inline src/lib.rs:5]: why are we enabling this?

**Review by jkvargas — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 3 above)
Comment (EYHN) [reply .github/workflows/release.yml:33]: This step compiles the static link library, only the static link feature is opened here

File: Cargo.toml (hunk 2 above)
Comment (EYHN) [reply Cargo.toml:22]: Yes, I just put it back.

**Comment by jkvargas** [comment]
hey bud, I didn't forget your PR.
I am moving to a new house, as soon as I get that sorted Ill come back to this :)

**Pull request merged**